tools/gromacs/insertmol.xml

```diff
@@ -0,0 +1,126 @@
+<tool id="gmx_insertmolecules" name="GROMACS insert-molecules" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
+    <description>to structure and topology files</description>
+    <macros>
+        <import>macros.xml</import>
+        <token name="@GALAXY_VERSION@">0</token>
+    </macros>
+
+    <expand macro="requirements">
+       <requirement type="package" version="3.7">python</requirement>
+    </expand>
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        ln -s '$gro_input' ./gro_input.gro &&
+        ln -s '$mol_input' ./mol_input.gro &&
+
+        gmx insert-molecules -f ./gro_input.gro -ci ./mol_input.gro -o multiples.gro -nmol '$nmols'
+          #if $str($box.config) == "true":
+              -box $box.dim
+          #end if
+           
+          >> verbose.txt 2>&1
+          && python '$__tool_directory__/nmols.py'
+    ]]></command>
+
+    <inputs>
+        <param argument="gro_input" type="data" format='gro' label="GRO structure file to insert into"/>
+        <param argument="mol_input" type="data" format='gro' label="Inserting molecule's GRO structure file"/>
+        <param name="nmols" type="integer" value="0" min="0" label="Number of copies of the molecule to add to system"/>
+        <conditional name="box">
+            <param name="config" type="select" label="Change box size?">
+                <option value="true">Yes</option>
+                <option value="false" selected="true">No</option>
+            </param>
+            <when value="true">
+                <param name="dim" type="integer" value="0" min="0" label="New box dimensions in nanometers. By default produces a cubic box, so the value provided is assigned to all 3 dimensions x, y, and z." help="This is useful if you're adding a lot of large molecules that may not fit inside the original box size. These dimensions, and box type can always be changed later with gmx_editconf." />
+            </when>
+            <when value="false"/>
+        </conditional>   
+    </inputs>
+    <outputs>
+        <data name="output1" format="gro" from_work_dir="./multiples.gro" label="GROMACS new structure with inserted-molecules (GRO) on ${on_string}"/>
+        <data name="output2" format="txt" from_work_dir="./addedmols.txt" label="Total number of molecules succesfully added on ${on_string}"/>
```

Comment:
Not really convinced about this, I guess the idea is that the user modifies the top file themselves? It would be nice if we could do this as part of the tool.

Reply:
Hey Simon, 

We were originally going to make this an all-in-one tool, but we decided to keep it as the standard gmx tool. We are working on a separate "add topologies" tool which will add topologies for inserted molecules, water, NaCL, and everything else. This output containing the number of added molecules would then be used as an input in the next step of adding topologies. 

The reason why this is necessary is because not all systems can be setup via the GROMACS initial setup tool (not all force fields are present in gromacs, which leads to unrecognized residue and atom types depending on your system). So having a separate tool that adds topologies will allow versatility for users.

Reply:
If I understand this correctly, the addedmols.txt will just contain a single integer, with the same value as the nmols parameter? If so, I don't think this is useful, I think the user can take care of this parameter themselves and remembering to use the same value for both tools.